### PR TITLE
Updating API version of admissionregistration.k8s.io

### LINKED ISSUE #1265: CRDs and webhook are registered using v1beta1 of the respective K8s API (both to be desupported in K8s 1.22)
The webhook is still registered admissionregistration.k8s.io/v1beta1 API version of MutatingWebhookConfiguration, which will hbe desupported with K8s 1.22, see https://kubernetes.io/docs/reference/using-api/deprecation-guide/#webhook-resources-v122.

This should be updated soon, or the use of the webhook should be abandoned, as proposed in https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/issues/1176 (which seems to make a lot of sense, considering that the webhook seems to be the cause of many issues).

A similar issue exists with the CRDs, which use apiextensions.k8s.io/v1beta1, also to be desupported with K8s 1.22:
https://kubernetes.io/docs/reference/using-api/deprecation-guide/#customresourcedefinition-v122

### LINKED ISSUE #1339: Operator projects using the removed APIs in k8s 1.22 requires changes.
### Problem Description

Kubernetes has been deprecating API(s), which will be removed and are no longer available in 1.22. Operators projects using [these APIs](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-22) versions will **not** work on Kubernetes 1.22 or any cluster vendor using this Kubernetes version(1.22), such as OpenShift 4.9+. Following the APIs that are most likely your projects to be affected by:
- apiextensions.k8s.io/v1beta1: (Used for CRDs and available since v1.16)
- rbac.authorization.k8s.io/v1beta1: (Used for RBAC/rules and available since v1.8)
- admissionregistration.k8s.io/v1beta1 (Used for Webhooks and available since v1.16)

Therefore, looks like this project distributes solutions in the [repository](https://github.com/redhat-openshift-ecosystem/community-operators-prod/tree/main/operators/spark-gcp) and does not contain any version compatible with k8s 1.22/OCP 4.9. ([More info](https://github.com/redhat-openshift-ecosystem/community-operators-prod/discussions/138)). Following some findings by checking the distributions published:

- **sparkoperator.v2.4.0**: this distribution is using APIs which were deprecated and removed in v1.22. More info: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-22. Migrate the API(s) for CRD: (["scheduledsparkapplications.sparkoperator.k8s.io" "sparkapplications.sparkoperator.k8s.io"]) 

**NOTE:** The above findings are only about the manifests shipped inside of the distribution. It is not checking the codebase.

### How to solve

It would be very nice to see new distributions of this project that are no longer using [these APIs](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-22) and so they can work on Kubernetes 1.22 and newer and published in the [community-operators](https://github.com/redhat-openshift-ecosystem/community-operators-prod/tree/main/operators/spark-gcp) collection. OpenShift 4.9, for example, will not ship operators anymore that do still use v1beta1 extension APIs.

Due to the number of options available to build Operators, it is hard to provide direct guidance on updating your operator to support Kubernetes 1.22. Recent versions of the [OperatorSDK](https://github.com/operator-framework/operator-sdk) greater than [1.0.0](https://github.com/operator-framework/operator-sdk/releases) and [Kubebuilder](https://github.com/kubernetes-sigs/kubebuilder) greater than [3.0.0](https://github.com/kubernetes-sigs/kubebuilder/releases) scaffold your project with the latest versions of these APIs (all that is generated by tools only). See the guides to upgrade your projects with OperatorSDK [Golang](https://sdk.operatorframework.io/docs/building-operators/golang/migration/), [Ansible](https://sdk.operatorframework.io/docs/building-operators/ansible/migration/), [Helm](https://sdk.operatorframework.io/docs/building-operators/helm/migration/) or the [Kubebuilder](https://book.kubebuilder.io/migration/v2vsv3.html) one. For APIs other than the ones mentioned above, you will have to check your code for usage of removed API versions and upgrade to newer APIs. The details of this depend on your codebase.

**If this projects only need to migrate the API for CRDs and it was built with [OperatorSDK](https://github.com/operator-framework/operator-sdk) versions lower than [1.0.0](https://github.com/operator-framework/operator-sdk/releases) then, you maybe able to solve it with an [OperatorSDK](https://github.com/operator-framework/operator-sdk) version >= v0.18.x < 1.0.0:**

> $ operator-sdk generate crds --crd-version=v1
> INFO[0000] Running CRD generator.                       
> INFO[0000] CRD generation complete.

**Alternatively, you can try to upgrade your manifests with [controller-gen](https://book.kubebuilder.io/reference/controller-gen.html) ([version >= v0.4.1](https://github.com/kubernetes-sigs/controller-tools/releases/tag/v0.4.1)) :**

#### If this project does not use Webhooks:

> $ controller-gen crd:trivialVersions=true,preserveUnknownFields=false rbac:roleName=manager-role  paths="./..."

#### If this project is using Webhooks:

1. Add the markers [sideEffects](https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/webhook/parser.go#L81) and [admissionReviewVersions](https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/webhook/parser.go#L114) to your webhook (Example with sideEffects=None and admissionReviewVersions={v1,v1beta1}: [memcached-operator/api/v1alpha1/memcached_webhook.go](https://github.com/operator-framework/operator-sdk/blob/master/testdata/go/v3/memcached-operator/api/v1alpha1/memcached_webhook.go#L39)):

2. Run the command:
> $ controller-gen crd:trivialVersions=true,preserveUnknownFields=false rbac:roleName=manager-role webhook paths="./..."

For further information and tips see the [comment](https://github.com/redhat-openshift-ecosystem/community-operators-prod/discussions/138#discussioncomment-1245759).

### PR DESCRIPTION
**Pull request by sairamankumar2**
This PR upgrades API version of admission registration objects of kind MutatingWebhook and ValidatingWebhook to v1 from v1beta1. v1beta1 version of admission registration is not [supported](https://kubernetes.io/blog/2021/08/04/kubernetes-1-22-release-announcement/#removal-of-several-deprecated-beta-apis) in 1.22+ version of Kubernetes cluster. 

The PR closes #1265 and #1375
With some additional work can fix #1339

### CODE REVIEW

**Comment by sairamankumar2** [comment]
As this version of operator contains breaking changes, should we also modify the operator version and make other changes to docs?

**Comment by liyinan926** [comment]
Thanks for the PR! I think we should also modify the operator version and make relevant doc changes. I can cut a new release afterwards.

**Comment by sairamankumar2** [comment]
Thanks @liyinan926,
Can you please list the files/docs that we need to modify for this release. Can i take this PR as reference #1153 for making the initial changes?

**Comment by sairamankumar2** [comment]
Hi @liyinan926,
Have updated the operator version to `v1beta2-1.3.0-3.1.1`. Kindly let me know if I have missed something.

File: README.md [hunk 1]
```diff
@@ -22,6 +22,8 @@ Customization of Spark pods, e.g., mounting arbitrary volumes and setting pod af
 
 * Version >= 1.13 of Kubernetes to use the [`subresource` support for CustomResourceDefinitions](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#subresources), which became beta in 1.13 and is enabled by default in 1.13 and higher.
 
+* Version >= 1.16 of Kubernetes to use the MutatingWebhook/v1 and ValidatingWebhook/v1 of admissionregistration.k8s.io
```
Comment (liyinan926) [inline README.md:25]: nit: to use `MutatingWebhook ` and `ValidatingWebhook ` `admissionregistration.k8s.io/v1`.

Comment (sairamankumar2) [reply README.md]: Fixed the typo

File: README.md [hunk 2]
```diff
@@ -22,6 +22,8 @@ Customization of Spark pods, e.g., mounting arbitrary volumes and setting pod af
 
 * Version >= 1.13 of Kubernetes to use the [`subresource` support for CustomResourceDefinitions](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#subresources), which became beta in 1.13 and is enabled by default in 1.13 and higher.
 
+* Version >= 1.16 of Kubernetes to use the MutatingWebhook and ValidatingWebhook of admissionregistration.k8s.io/v1
```
Comment (liyinan926) [inline README.md:25]: Nit: please add a period at the end, and use "`" around the k8s resources.

Comment (sairamankumar2) [reply README.md]: Wrapped k8s resources in code block and also added `apiVersion:` before admissionregistration.k8s.io/v1.

**Review by liyinan926 — APPROVED** [review]
LGTM

**Comment by sp-matrix** [comment]
@liyinan926 @sairamankumar2 : It would be great, if you can update the OperatorHub with the latest version. I have been asked to update the operator from version 2.4 to the version that will be supported by openshift 4.9

**Pull request merged**